### ISSUE TITLE
Parallel training a single model – am I doing it right?

### ISSUE THREAD

**Issue by radrow** [horovod/horovod#980]
I would like to train some model on 2 separate GPUs, so I have followed all the steps mentioned in the README and ran
```
horovodrun -np 2 -H localhost:2 python train.py
```

`train.py` basically loads dataset from disk, constructs tf model and runs a training loop. It initializes hvd and wraps optimizer as mentioned in the instructions.

However, this command runs two separate python processes that seem to be completely independent – according to printed logs each loads the dataset separately, builds a model and  runs training on its own. Is it how it is supposed to look like? I don't observe any speedup – it is even worse with `-np 2`. I just looks like each log is duplicated, I cannot see any connection between these two instances

**Comment by tgaddair** [comment]
Hey @radrow, Horovod does indeed run multiple Python processes.  The coordination between them only comes in during the backward pass when gradients are aggregated together via the `DistributedOptimizer`.

The way you achieve speedup is by distributing the data between the workers.  So in your case since you have two processes each process should only be processing half the total number of batches in each epoch. Take a look at our [examples](https://github.com/horovod/horovod/tree/master/examples) to see how to do this in practice.

**Comment by radrow** [comment]
Thanks, that solves my doubts
